daemon/container_operations.go

```diff
@@ -757,6 +757,10 @@ func (daemon *Daemon) connectToNetwork(container *container.Container, idOrName
 
 	container.NetworkSettings.Ports = getPortMapInfo(sb)
 
+	if err := container.ToDisk(); err != nil {
```

Comment:
Yes, the reason that changing from `ToDiskLocking()` to `ToDisk()` is because `ToDiskLocking()` leads a dead lock blocking the test. 

Yes, I am still on it to dig further. Moving to WIP.